framework/src/ble_manager/ble_manager_state.c

```diff
@@ -125,7 +125,9 @@ static void _event_caller(int evt_pri, void *data) {
 		} break;
 		case BLE_EVT_CLIENT_DISCONNECT: {
 			ble_client_device_disconnected_cb callback = msg->param[0];
-			callback(msg->param[1]);
+			uint32_t value = *(uint32_t *)msg->param[2];
+			uint8_t reason = (uint8_t)(value & 0xff);
```

Comment:
We could combine the above two lines into a single statement.

Reply:
To represent the meaning of data clearly, I separated it into 2 parts